src/profile/utilities.jl

```diff
@@ -0,0 +1,165 @@
+    positions[:σs] = (lastpos + 1):(lastpos + sz)
+    lastpos += sz
+    corrpos = NTuple{3,Int}[]
+    for (i, re) in enumerate(m.reterms)
+        (isa(re.λ, Diagonal) || isa(re, ReMat{T,1})) && continue
+        indm = indmat(re)
+        for j in axes(indm, 1)
+            rowj = view(indm, j, :)
+            for k in (j + 1):size(indm, 1)
+                if !iszero(dot(rowj, view(indm, k, :)))
+                    push!(corrpos, (i, j, k))
+                end
+            end
+        end
+    end
+    sz = length(corrpos)
+    if sz > 0
+        append!(nmvec, _generatesyms('ρ', sz))
+        positions[:ρs] = (lastpos + 1):(lastpos + sz)
+        lastpos += sz
+    end
+    sz = length(m.θ)
+    append!(nmvec, _generatesyms('θ', sz))
+    positions[:θ] = (lastpos + 1):(lastpos + sz)
+    return TableColumns((nmvec...,), positions, zeros(T, length(nmvec)), corrpos)
+end
+
+function mkrow!(tc::TableColumns{T,N}, m::LinearMixedModel{T}, ζ::T) where {T,N}
+    (; cnames, positions, v, corrpos) = tc
+    v[1] = ζ
+    fixef!(view(v, positions[:β]), m)
+    v[first(positions[:σ])] = m.σ
+    σvals!(view(v, positions[:σs]), m)
+    getθ!(view(v, positions[:θ]), m)  # must do this first to preserve a copy
+    if length(corrpos) > 0
+        ρvals!(view(v, positions[:ρs]), corrpos, m)
+        setθ!(m, view(v, positions[:θ]))
+    end
+    return NamedTuple{cnames,NTuple{N,T}}((v...,))
+end
+
+function parsej(sym::Symbol) # return the index from symbol names like :θ1, :θ01, etc.
+    symstr = string(sym)                                     # convert Symbol to a String
+    return parse(Int, SubString(symstr, nextind(symstr, 1))) # drop first Unicode character and parse as Int
+end
+
+#=  # It appears that this method is not used
+"""
+    σvals(m::LinearMixedModel)
+
+Return a Tuple of the standard deviation estimates of the random effects
+"""
+function σvals(m::LinearMixedModel{T}) where {T}
+    (; σ, reterms) = m
+    isone(length(reterms)) && return σvals(only(reterms), σ)
+    return (collect(Iterators.flatten(σvals.(reterms, σ)))...,)
+end
+=#
+
+function σvals!(v::AbstractVector{T}, m::LinearMixedModel{T}) where {T}
+    (; σ, reterms) = m
+    isone(length(reterms)) && return σvals!(v, only(reterms), σ)
+    ind = firstindex(v)
+    for t in m.reterms
+        S = size(t.λ, 1)
+        σvals!(view(v, ind:(ind + S - 1)), t, σ)
+        ind += S
+    end
+    return v
+end
+
+function ρvals!(
+    v::AbstractVector{T}, corrpos::Vector{NTuple{3,Int}}, m::LinearMixedModel{T}
+) where {T}
+    reterms = m.reterms
+    lasti = 1
+    λ = first(reterms).λ
+    for r in eachrow(λ)
+        normalize!(r)
+    end
+    for (ii, pos) in enumerate(corrpos)
+        i, j, k = pos
+        if lasti ≠ i
+            λ = reterms[i].λ
+            for r in eachrow(λ)
+                normalize!(r)
+            end
+            lasti = i
+        end
+        v[ii] = dot(view(λ, j, :), view(λ, k, :))
+    end
+    return v
+end
+
+"""
```

Comment:
I feel like this could be useful in a few other places where we need to nudge things away from the boundary for refitting, but that's a follow-up for later